src/lib/searchUtilActions/SearchRepositoryHelper.ts

```diff
@@ -32,11 +48,18 @@ export async function getAasThumbnailFromAllRepos(
 ) {
     const basePathUrls = await getConnectionDataByTypeAction({ id: '0', typeName: 'AAS_REPOSITORY' });
 
-    for (const url of basePathUrls) {
-        const image = await repositoryClient.getThumbnailFromShell(aasId, undefined, url);
-        if (image.size != 0) {
+    const promises = basePathUrls.map((url) =>
+        repositoryClient.getThumbnailFromShell(aasId, undefined, url).then((image) => {
+            if (image.size === 0) {
+                throw new Error('Empty image');
+            }
             return image;
-        }
+        }),
+    );
+
+    try {
+        return await Promise.any(promises);
```

Comment:
Just to make sure:
 This code will only throw, if ALL `promises` throw an error?
 Else we want to take the first valid one, i think.